### PR TITLE
Adds achievement datums

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus, Willox & AnturK
 add: Achievements are now datums
/:cl:

Changes medals into achievements, which allows us to be more conservative with our polling to the hub by only sending over achievement data once every 5 minutes (and on SS shutdown). Besides that, if we move to our own solution (database) we can easily migrate.

image
heres an image of the current achievement browser. Its pretty basic. Down the line we could link achievements being unlocked to other things such as cool visual unlocks but we're better off doign that once we move away from the Byond medal page.

I also realized that most of our medals aren't even in the hub page, most likely because the only person who has access to my knowledge is @MrStonedOne  (ShadowLight left)

### CODE REVIEW

**Comment by Qustinnus** [comment]
Also testing this is not possible for me on anything but live server due to the hub page requirements.

**Comment by MCterra10** [comment]
Minecraft style achievements in SS13 when

**Comment by Qustinnus** [comment]
@MCterra10 done

**Comment by MCterra10** [comment]
Absolutely based

**Comment by Whoneedspacee** [comment]
we goon now bois thanks so much man

File: code/datums/achievements/boss_achievements.dm [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+/datum/achievement/colussus_kill
+	name = "Colussus Killer"
+	desc = "The bigger they are.... the better the loot"
+	achievement_type = ACHIEVEMENT_DEFAULT
+	hub_id = BOSS_MEDAL_COLOSSUS
+
+/datum/achievement/drake_kill
+	name = "Blood-drunk Miner Killer"
+	desc = "I guess he couldn't handle his drink that well."
```
Comment (Fikou) [inline code/datums/achievements/boss_achievements.dm]: you called the ash drake achievement blood drunk miner

Comment (Fikou) [reply code/datums/achievements/boss_achievements.dm]: also you forgot drake crusher

File: code/datums/achievements/boss_achievements.dm [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+/datum/achievement/colussus_crusher
+	name = "Colussus Crusher"
+	desc = "The bigger they are.... the better the loot"
+	achievement_type = ACHIEVEMENT_DEFAULT
+	hub_id = BOSS_MEDAL_COLOSSUS_CRUSHER
+
+/datum/achievement/drake_crusher
+	name = "Blood-drunk Miner Crusher"
+	desc = "I guess he couldn't handle his drink that well."
```
Comment (Fikou) [inline code/datums/achievements/boss_achievements.dm]: ditto

**Comment by Kmc2000** [comment]
https://i.imgur.com/Hqlitbz.png

**Comment by Ziiro** [comment]
Might be cool to attach species, jobs, and quirks to getting certain achievements down the road.

**Comment by Qustinnus** [comment]
@Ziiro wdym

**Comment by Ziiro** [comment]
Things like:

- If the player has [some ridiculous amount]u of felinid toxin in their system, they get the achievement "Degenerate" and permanently unlock Felinids as a species option

- Dying to electrocution in 10 separate games gives the achievement "Shockingly bad idea" unlocks Ethereal

- Offering 10 corpses to the Ashwalker tendril gives an achievement that unlocks lizards,

etc

**Comment by Qustinnus** [comment]
sounds like it'll make people angry

**Comment by Whoneedspacee** [comment]
uh no

achievements shouldnt unlock content like that

bragging rights cosmetics though hell yes goon did that great with different lab coat colors and shit

**Comment by Ziiro** [comment]
Whoneedspacee has better ideas but that's what I meant yeah

**Comment by optimumtact** [comment]
I think it's time for a new hub, with the access details stored on coderbus private pages

**Comment by Ty-the-Smonk** [comment]
station round based Minecraft achievements should exist and do something

**Comment by 81Denton** [comment]
conflicts

**Comment by Qustinnus** [comment]
image

image of achievement menu

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/achievements/boss_achievements.dm [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	hub_id = BOSS_MEDAL_DRAKE_CRUSHER
+
+/datum/award/achievement/boss/hierophant_crusher
+	name = "Hierophant Crusher"
+	desc = "Hierophant, but not triumphant."
+	hub_id = BOSS_MEDAL_HIEROPHANT_CRUSHER
+
+/datum/award/achievement/boss/legion_crusher
+	name = "Legion Crusher"
+	desc = "We were many..now we are none."
```
Comment (Fikou) [inline code/datums/achievements/boss_achievements.dm]: add a space and third dot

Comment (Qustinnus) [reply code/datums/achievements/boss_achievements.dm]: no

File: code/datums/achievements/misc_achievements.dm [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+/datum/award/achievement/misc/time_waste
+	name = "Time waster"
+	desc = "Speak no evil, hear no evil, see just errors"
+	hub_id = MEDAL_TIMEWASTE
+
+/datum/award/achievement/misc/feat_of_strength
+	name = "Feat of Strength"
+	desc = "If the rod is immovable, is it passing you or are you passing it?"
+	hub_id = MEDAL_TIMEWASTE
```
Comment (Fikou) [inline code/datums/achievements/misc_achievements.dm]: wrong achievement

File: code/datums/achievements/boss_achievements.dm (hunk 3 above)
Comment (Fikou) [reply code/datums/achievements/boss_achievements.dm]: but muh space gramar

**Review by stylemistake — APPROVED** [review]
UI looks fine to me. Just needs a rebuild, one more time.

File: code/__DEFINES/achievements.dm [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+#define BOSS_MEDAL_TENDRIL	  "Tendril Exterminator"
+#define BOSS_MEDAL_SWARMERS   "Swarmer Beacon Killer"
+
+#define BOSS_MEDAL_MINER_CRUSHER	  "Blood-drunk Miner Crusher"
+#define BOSS_MEDAL_BUBBLEGUM_CRUSHER  "Bubblegum Crusher"
+#define BOSS_MEDAL_COLOSSUS_CRUSHER	  "Colossus Crusher"
+#define BOSS_MEDAL_DRAKE_CRUSHER	  "Drake Crusher"
+#define BOSS_MEDAL_HIEROPHANT_CRUSHER "Hierophant Crusher"
+#define BOSS_MEDAL_LEGION_CRUSHER	  "Legion Crusher"
+#define BOSS_MEDAL_SWARMERS_CRUSHER  "Swarmer Beacon Crusher"
```
Comment (actioninja) [inline code/__DEFINES/achievements.dm]: if you're going to align these align them properly

File: code/__DEFINES/achievements.dm [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+//Misc Medal hub IDs
+#define MEDAL_METEOR 	"Your Life Before Your Eyes"
+#define MEDAL_PULSE 	"Jackpot"
+#define MEDAL_TIMEWASTE "Overextended The Joke"
+#define MEDAL_RODSUPLEX "Feat of Strength"
+#define MEDAL_CLOWNCARKING "Round and Full"
+#define MEDAL_THANKSALOT "The Best Driver"
+#define MEDAL_HELBITALJANKEN "Hel-bent on Winning"
+#define MEDAL_MATERIALCRAFT "Getting an Upgrade"
+
```
Comment (actioninja) [inline code/__DEFINES/achievements.dm]: either no alignment or do it right

**Review by actioninja — CHANGES_REQUESTED** [review]
this isn't a full review but this irritates me

File: tgui-next/packages/tgui/routes.js [hunk 7]
```diff
@@ -32,14 +31,10 @@ import { VaultController } from './interfaces/VaultController';
...
 const ROUTES = {
-  acclimator: {
-    component: () => Acclimator,
-    scrollable: false,
-  },
   achievements: {
     component: () => Achievements,
-    scrollable: true,
   },
+  scrollable: true,
```
Comment (stylemistake) [inline tgui-next/packages/tgui/routes.js]: bruh

**Pull request merged**